src/drivers/infiniband/hermon.c

```diff
@@ -2097,6 +2219,7 @@ static void hermon_poll_eq ( struct ib_device *ibdev ) {
 		/* Ring doorbell */
 		MLX_FILL_1 ( &db_reg.event, 0,
 			     ci, ( hermon_eq->next_idx & 0x00ffffffUL ) );
+		barrier();
```

Comment:
I don't see any memory barrier in `hermon_poll_cq()`.  Did you mean `hermon_post_send()`?  If so then:

- the `barrier()` before `writel()` in `hermon_post_send()` is to ensure that the WQE is flushed before ringing the doorbell
- no similar requirement exists in `hermon_poll_eq()` since it does not modify DMA-accessible host memory: the EQEs are read-only as far as software is concerned.

Reply:
I misread the code - patch withdrawn.